### PR TITLE
Add OS support tables

### PR DESCRIPTION
**Pull request by SueChaplain**
Adding operating system support to the
support statement.

Closes: #41 

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

**Comment by SueChaplain** [comment]
Jenkins doc stage

File: docs/openj9_support.md [hunk 1]
```diff
@@ -72,26 +71,93 @@ show the minimum build configurations that we have tested to build an OpenJDK bi
...
+|---------------------------------------|--------|--------|------------|------|------|
+| Centos 6                              |   Y    |   Y    |     Y      |   N  |  N   |
+| Centos 7                              |   Y    |   Y    |     Y      |   N  |  N   |
+| Red Hat Enterprise Linux (RHEL) 6     |   Y    |   Y    |     Y      |   Y  |  Y   |
+| RHEL 7                                |   Y    |   Y    |     Y      |   Y  |  Y   |
+| SUSE Linux Enterprise Server (SLES) 12|   Y    |   Y    |     Y      |   Y  |  Y   |
+| Ubuntu 16.04                          |   Y    |   Y    |     Y      |   N  |  Y   |
+| Ubuntu 18.04                          |   Y    |   Y    |     Y      |   N  |  Y   |
+
+<i class="fa fa-pencil-square-o" aria-hidden="true"></i><span class="sr-only">Note</span> **Note:** Not all of these distributions are tested, but Linux distributions that have a
```
Comment (pshipton) [inline docs/openj9_support.md]: This shows Note twice, with the second occurrence in bold.

Comment (pshipton) [reply docs/openj9_support.md]: Above OpenJDK 8 there is text which should be removed "The following tables show the minimum build configurations that we have tested to build an OpenJDK binary."

File: docs/openj9_support.md [hunk 2]
```diff
@@ -72,26 +71,93 @@ show the minimum build configurations that we have tested to build an OpenJDK bi
...
+minimum glibc version 2.12 are expected to function without problems.
+
+| Windows                    |  x32   |  x64  |
+|----------------------------|--------|-------|
+| Windows 10                 |   Y    |   Y   |
+| Windows Server 2012        |   N    |   Y   |
+| Windows Server 2012 R2     |   N    |   Y   |
+| Windows Server 2016        |   N    |   Y   |
+
```
Comment (pshipton) [inline docs/openj9_support.md]: Why the limitation of 32-bit on Servers OSes? Its a bit weird we use Server to build 32-bit but its not supported to run there.

File: docs/openj9_support.md [hunk 3]
```diff
@@ -72,26 +71,93 @@ show the minimum build configurations that we have tested to build an OpenJDK bi
...
+#### OpenJDK 8 build environment
+
+The following table shows the operating system and compilers that the project use to build and test OpenJ9 for different platforms:
 
 | Platform                                    | Operating system          |  Compiler                       |  Comments      |
 |---------------------------------------------|---------------------------|---------------------------------|----------------|
-| Linux x86 64-bit                            | Ubuntu 14.04              | gcc 4.4.7                       |                |
-| Linux on POWER<sup>&reg;</sup> LE 64-bit    | Ubuntu 14.04              | gcc 4.4.7                       |                |
-| Linux on IBM Z<sup>&reg;</sup> 64-bit       | Ubuntu 14.04              | gcc 4.4.7                       |                |
+| Linux x86 64-bit                            | Ubuntu 16.04              | gcc 4.4.7                       |                |
```
Comment (pshipton) [inline docs/openj9_support.md]: Now this is a build table, it should be gcc 4.8.5

File: docs/openj9_support.md [hunk 4]
```diff
@@ -72,26 +71,93 @@ show the minimum build configurations that we have tested to build an OpenJDK bi
...
+| Linux on POWER<sup>&reg;</sup> LE 64-bit    | Ubuntu 16.04              | gcc 4.4.7                       |                |
+| Linux on IBM Z<sup>&reg;</sup> 64-bit       | Ubuntu 16.04              | gcc 4.4.7                       |                |
 | Windows x86 32-bit                          | Windows Server 2012       | Microsoft Visual Studio 2010 SP1|                |
```
Comment (pshipton) [inline docs/openj9_support.md]: Either this or the 9/10 table is incorrect for the OS. We use the same machines to build 8/9/10.

File: docs/openj9_support.md [hunk 5]
```diff
@@ -117,9 +213,3 @@ When public support for an operating system version ends, OpenJ9 can no longer b
 | Linux on IBM Z 64-bit       | Ubuntu 16.04             | gcc 4.8.5                       |                |
```
Comment (pshipton) [inline docs/openj9_support.md]: This line in the JDK9 environment shows the kernel level in brackets, which I don't think is required.

File: docs/openj9_support.md (hunk 4 above)
Comment (pshipton) [reply docs/openj9_support.md]: According to the machine labels https://ci.eclipse.org/openj9/, they are all Server 2012 R2 machine so the tables for 9/10 should be updated to reflect this.

File: docs/openj9_support.md [hunk 6]
```diff
@@ -72,26 +71,93 @@ show the minimum build configurations that we have tested to build an OpenJDK bi
...
+|--------------|----------|----------|
+| AIX 7.1 TL4  |    Y     |    Y     |
+| AIX 7.2      |    Y     |    Y     |
 
 When public support for an operating system version ends, OpenJ9 can no longer be supported on that level.
 
-### OpenJDK 9 build environment
+#### OpenJDK 9 build environment
+
+The following table shows the operating system and compilers that the project use to build and test OpenJ9 for different platforms:
```
Comment (pshipton) [inline docs/openj9_support.md]: This Linux x86 line in the table shows the kernel level in brackets, which I don't think is required.

File: docs/openj9_support.md [hunk 7]
```diff
@@ -102,13 +168,43 @@ When public support for an operating system version ends, OpenJ9 can no longer b
 | AIX POWER BE 64-bit         | AIX 7.1 TL04             | xlc/C++ 13.1.3                  |                |
 
 
-OpenJDK9 binaries are supported on the minimum operating system levels shown in the table.
-For Linux, other distributions are supported provided that they have a Linux kernel v4.4 or later
-(Look in the `/proc/version` file to find the kernel version number).
+### OpenJDK 10
+
+OpenJDK8 binaries are supported on the minimum operating system levels shown in the following tables:
```
Comment (pshipton) [inline docs/openj9_support.md]: Says OpenJDK8 but this is the section for 10.

File: docs/openj9_support.md (hunk 1 above)
Comment (SueChaplain) [reply docs/openj9_support.md]: Thanks for reviewing. I should get in the habit of using WIP in the title. 

The "Note twice" is only an issue when viewed in GitHub, not on the staging site. GitHub does not understand the HTML <span> so repeats the word.

Removed the text from the top paragraph as you pointed out.

File: docs/openj9_support.md (hunk 2 above)
Comment (SueChaplain) [reply docs/openj9_support.md]: That was something I was going to ask you about. Fixed my guess.

File: docs/openj9_support.md (hunk 6 above)
Comment (SueChaplain) [reply docs/openj9_support.md]: Fixed

File: docs/openj9_support.md (hunk 7 above)
Comment (SueChaplain) [reply docs/openj9_support.md]: Fixed

File: docs/openj9_support.md (hunk 3 above)
Comment (SueChaplain) [reply docs/openj9_support.md]: Fixed and removed the note below the table to say we'll be moving to 4.8 soon.

File: docs/openj9_support.md (hunk 4 above)
Comment (SueChaplain) [reply docs/openj9_support.md]: Changed all the Windows machines to be Windows Server 2012 R2. Hope that was what you intended.

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by pshipton** [comment]
We still need to address https://github.com/eclipse/openj9-docs/issues/40
Although OpenJ9 uses 4.8.5 to build, so its appropriate to show 4.8.5 in the table of what is tested, we still want to state that 4.4.7 is the minimum level and the code must compiled on this level for the time being.

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
@pshipton Added back in the compiler level warning for OpenJDK 8, so hopefully this is good to go (when we are ready with 0.9.0). If you look at this again, you will notice that I reordered the topic. So we have the supported platforms (8,9,10) and then the build environments (8,9,10), mostly because of a formatting problem I could not solve when the file was building. The changes can be seen on the staging server if you want to take a look.

File: docs/openj9_support.md [hunk 8]
```diff
@@ -43,20 +43,19 @@ supported by any open project.
...
 |-----------------|--------------------|-----------|------|-------|-------------|-------|-------|
 | v0.8.0          | March 2018         | Yes       | No   |       |             |       |       |
-| v0.9.0          | May 2018 (\*1)     | Yes       | No   | Yes   |             |       |       |
-| v0.10.0         | August 2018 (\*2)  | Yes       | No   | Yes   |             |       |       |
-| v0.11.0         | October 2018 (\*2) | Yes       | No   | No    | Yes         |       |       |
-| v0.12.0         | January 2019 (\*2) | Yes       | No   | No    | Yes         |       |       |
-| v0.13.0         | April 2019 (\*2)   | No (\*3)  | No   | No    | No (\*3)    | Yes   |       |
-| v0.14.0         | July 2019 (\*2)    | No (\*3)  | No   | No    | No (\*3)    | Yes   |       |
-| v0.15.0         | October 2019 (\*2) | No (\*3)  | No   | No    | No (\*3)    | No    | Yes   |
+| v0.9.0          | May 2018           | Yes       | No   | Yes   |             |       |       |
```
Comment (pshipton) [inline docs/openj9_support.md]: Should we update to June since the release is late?

**Comment by pshipton** [comment]
Other than the 0.9.0 release date, lgtm.
@DanHeidinga @mstoodle any concerns?

**Comment by mstoodle** [comment]
No concerns from me. +1 and thanks for pulling it together!

File: docs/openj9_support.md [hunk 9]
```diff
@@ -66,60 +65,147 @@ For any issues or limitations of an Eclipse OpenJ9 release, read the [release no
...
+| RHEL 7                                |   Y    |   Y    |     Y      |   Y  |  Y   |
+| SUSE Linux Enterprise Server (SLES) 12|   Y    |   Y    |     Y      |   Y  |  Y   |
+| Ubuntu 16.04                          |   Y    |   Y    |     Y      |   N  |  Y   |
+| Ubuntu 18.04                          |   Y    |   Y    |     Y      |   N  |  Y   |
+
+<i class="fa fa-pencil-square-o" aria-hidden="true"></i><span class="sr-only">Note</span> **Note:** Not all of these distributions are tested, but Linux distributions that have a
+minimum glibc version 2.12 are expected to function without problems.
+
+| Windows                    |  x32   |  x64   |
+|----------------------------|--------|--------|
```
Comment (DanHeidinga) [inline docs/openj9_support.md]: 32 bit windows was only intended to be supported for JDK8.  No effort has been put into making it function for new JDK levels

File: docs/openj9_support.md [hunk 10]
```diff
@@ -66,60 +65,147 @@ For any issues or limitations of an Eclipse OpenJ9 release, read the [release no
...
+| SUSE Linux Enterprise Server (SLES) 12|   Y    |   Y    |     Y      |   Y  |  Y   |
+| Ubuntu 16.04                          |   Y    |   Y    |     Y      |   N  |  Y   |
+| Ubuntu 18.04                          |   Y    |   Y    |     Y      |   N  |  Y   |
+
+<i class="fa fa-pencil-square-o" aria-hidden="true"></i><span class="sr-only">Note</span> **Note:** Not all of these distributions are tested, but Linux distributions that have a
+minimum glibc version 2.12 are expected to function without problems.
+
+| Windows                    |  x32   |  x64  |
+|----------------------------|--------|-------|
+| Windows 10                 |   Y    |   Y   |
```
Comment (DanHeidinga) [inline docs/openj9_support.md]: 32 bit windows was only intended to be supported for JDK8. No effort has been put into making it function for new JDK levels

File: docs/openj9_support.md [hunk 11]
```diff
@@ -66,60 +65,147 @@ For any issues or limitations of an Eclipse OpenJ9 release, read the [release no
...
+| Windows Server 2012        |   Y    |   Y   |
+| Windows Server 2012 R2     |   Y    |   Y   |
+| Windows Server 2016        |   Y    |   Y   |
+
+
+| AIX          |  ppc32   | ppc64    |
+|--------------|----------|----------|
```
Comment (DanHeidinga) [inline docs/openj9_support.md]: Please remove 32 bit support from here as well.

File: docs/openj9_support.md [hunk 12]
```diff
@@ -66,60 +65,147 @@ For any issues or limitations of an Eclipse OpenJ9 release, read the [release no
...
+|----------------------------|--------|--------|
+| Windows 10                 |   Y    |   Y    |
+| Windows Server 2012        |   Y    |   Y    |
+| Windows Server 2012 R2     |   Y    |   Y    |
+| Windows Server 2016        |   Y    |   Y    |
+
+
+| AIX          | ppc32    |  ppc64   |
+|--------------|----------|----------|
+| AIX 7.1 TL4  |    Y     |    Y     |
```
Comment (DanHeidinga) [inline docs/openj9_support.md]: Please remove 32 bit support from here

File: docs/openj9_support.md [hunk 13]
```diff
@@ -66,60 +65,147 @@ For any issues or limitations of an Eclipse OpenJ9 release, read the [release no
...
-OpenJDK9 binaries are supported on the minimum operating system levels shown in the table.
-For Linux, other distributions are supported provided that they have a Linux kernel v4.4 or later
-(Look in the `/proc/version` file to find the kernel version number).
+### OpenJDK 10
+
+OpenJDK10 binaries are supported on the minimum operating system levels shown in the following tables:
+
+
+| Linux                                 | x32    |  x64   |  ppc64le   | Z31  | Z64  |
+|---------------------------------------|--------|--------|------------|------|------|
```
Comment (DanHeidinga) [inline docs/openj9_support.md]: Please remove the 32 bit support column.

File: docs/openj9_support.md [hunk 14]
```diff
@@ -66,60 +65,147 @@ For any issues or limitations of an Eclipse OpenJ9 release, read the [release no
...
-| Linux on IBM Z 64-bit       | Ubuntu 16.04             | gcc 4.8.5                       |                |
-| Windows x86 64-bit          | Windows Server 2016      | Microsoft Visual Studio 2013    |                |
-| AIX POWER BE 64-bit         | AIX 7.1 TL04             | xlc/C++ 13.1.3                  |                |
+### OpenJDK 9
+
+OpenJDK9 binaries are supported on the minimum operating system levels shown in the following tables:
+
+
+| Linux                                 |  x32   |  x64   |   ppc64le  | Z31  | Z64  |
+|---------------------------------------|--------|--------|------------|------|------|
```
Comment (DanHeidinga) [inline docs/openj9_support.md]: Please remove the 32 bit support column.

Comment (SueChaplain) [reply docs/openj9_support.md]: Done

File: docs/openj9_support.md (hunk 13 above)
Comment (SueChaplain) [reply docs/openj9_support.md]: Done

File: docs/openj9_support.md (hunk 12 above)
Comment (SueChaplain) [reply docs/openj9_support.md]: Done

File: docs/openj9_support.md (hunk 11 above)
Comment (SueChaplain) [reply docs/openj9_support.md]: Done

File: docs/openj9_support.md (hunk 9 above)
Comment (SueChaplain) [reply docs/openj9_support.md]: Done

**Comment by SueChaplain** [comment]
Hopefully all good now?

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by pshipton** [comment]
If you change the 0.9.0 release date to June then lgtm.

**Comment by pshipton** [comment]
Spoke (wrote) too soon. The Z31 column should be removed from JDK9+ as well.

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by pshipton** [comment]
Dan and I discussed some changes to the release predictions. Can you please make the following changes. We are inserting an additional release.

OpenJ9 release | Release date | JDK8 (LTS) | JDK9 | JDK10 | JDK11 (LTS) | JDK12 | JDK13
-- | -- | -- | -- | -- | -- | -- | --
v0.10.0 | July 2018 (*1) | Yes | No | Yes |   |   |  
v0.11.0 | Sept 2018 (*1) | Yes | No | No | Yes |   |  
v0.12.0 | Oct 2018 (*1) | Yes | No | No | Yes |   |  

and then v0.13.0 in January, etc.

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
Index topic updated with a "Support" section. 2 purposes:

1. Help people find the full support topic in the Reference section (partly address #38).
2. Add a Note about "z/OS" content in the user guide.

Also added JDK 11 into the top table, marked LTS.

**Comment by pshipton** [comment]
As per the OpenJ9 community meeting yesterday, we're not going to have an OpenJ9 0.10 release in July, this line should be removed. But you can change the 0.9.0 date to July since its not going to happen in June.

**Comment by SueChaplain** [comment]
The 0.10.0 line is removed and subsequent numbers updated.
Moved the 0.9.0 date to July.

**Comment by pshipton** [comment]
Any reason we can't go ahead and merge this?

**Comment by SueChaplain** [comment]
@pshipton  - I was holding it for 0.9.0 only because I had removed the note that 0.9.0 was an "expected release". If you're happy for this to go as is I will merge?

**Comment by pshipton** [comment]
I think its best to have the most current information delivered. I've merged so you don't need to merge your own PR.

**Pull request merged**